src/pool/cli.c

```diff
@@ -2125,6 +2189,8 @@ dc_pool_set_attr(tse_task_t *task)
 
 out:
 	tse_task_complete(task, rc);
+	if (new_names != NULL)
+		D_FREE(new_names);
```

Comment:
does new_names[i] always get free'd even when there is an error?

Reply:
my understanding is that tse_task_complete() will run all registered callbacks to the task, ie including the one for new_names[I]
and BTW this makes me think I should remove this D_FREE(new_names) as new_names has also its callback registered now, what do you think ?

Reply:
It does seem there might be a double-free there.

Reply:
I agree, but interesting is that NLT did not detect it apparently ??

Reply:
The fault injection testing is expensive (although easy to add new tests), it takes about ten minutes per test and currently only has two complete tests.

To add one can you tell me the name of a daos command that should be run, I assume it would be 

- create a container
- set two attributes on it
- run "daos container list-attr" as the test?

I can trivially add this for you, but not until after the school run today.